#files  nkx.fasta and pbx.fasta were generated by script 03_SF_annotation/05_grab_fasta_from_orthogroups.sh from proteinortho orthogroups
#alignment performed online on https://mafft.cbrc.jp/alignment/server/ with usage of following options:
#1. Use DASH to add homologous structures (Output original sequences only)
#2. Auto strategy
#3. Adding homologs with Mafft-homologs using UniRef50 (more comprehensive and requires longer search time) 
#these options chosen for better alignment of distant orthologs between Spiralia, Ecdysozoa and Vertebrata
#trimming performed with trimal (trimAl v1.4.rev15) in automated mode
trimal -automated1 -in aln_no_trimming/nkx2_aligned.fasta -out aln_trimmed/tr_nkx2_aligned.fasta
trimal -automated1 -in aln_no_trimming/pbx_aligned.fasta -out aln_trimmed/tr_pbx_aligned.fasta
#auto-selection of evolution model and phylogenetic tree construction conducted with IQTREE2 (IQ-TREE multicore version 2.2.2.3 COVID-edition for Linux 64-bit built May 11 2023)
#both for trimmed and untrimmed alignment, best variant was selected manually based on support values and biological correctness of the phylogenetic tree
#homo sapiens PBX and NKX-genes were used as outgroup
iqtree -s aln_no_trimming/nkx2_aligned.fasta -o NKX22_HUMAN,NKX25_HUMAN,NKX21_HUMAN -st AA -m TESTNEW -bb 1000 -alrt 1000 -abayes
iqtree -s aln_trimmed/tr_nkx2_aligned.fasta -o NKX22_HUMAN,NKX25_HUMAN,NKX21_HUMAN -st AA -m TESTNEW -bb 1000 -alrt 1000 -abayes
iqtree -s aln_no_trimming/pbx_aligned.fasta -o PBX1_HUMAN,PBX3_HUMAN,PBX2_HUMAN -st AA -m TESTNEW -bb 1000 -alrt 1000 -abayes
iqtree -s aln_trimmed/tr_pbx_aligned.fasta -o PBX1_HUMAN,PBX3_HUMAN,PBX2_HUMAN -st AA -m TESTNEW -bb 1000 -alrt 1000 -abayes
#for visualisation FigTree (v.1.4.4) was used 
