{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "808bbd4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pathlib\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed471ec2",
   "metadata": {},
   "source": [
    "### Save annotations for genes with homeodomains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ff143928",
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_name_in_row(annot_df, pattern):\n",
    "    return annot_df[annot_df.apply(lambda row: row.astype(str).str.contains(pattern, case=False).any(), axis=1)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a9775990",
   "metadata": {},
   "outputs": [],
   "source": [
    "worms = ['Pele','Pdum']\n",
    "for worm in worms:\n",
    "    combine_annotation_file = f'C:/worms/new/combined_annotations/{worm}_combined_annotation.csv'\n",
    "    annotation_df = pd.read_csv(combine_annotation_file, sep='\\t')\n",
    "    find_name_in_row(annotation_df, 'Homeodomain|Homeobox')\\\n",
    "        .to_csv(f'C:/worms/new/{worm}_homeobox_annotations.csv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "217aedcf",
   "metadata": {},
   "outputs": [],
   "source": [
    "p = pathlib.Path(r'C:\\worms\\new\\pbx_like\\trimmed_aligned_PBX_Spiralia.fasta')\n",
    "fasta_dict = {}\n",
    "with open(p,'r') as fasta_file:\n",
    "    for line in fasta_file:\n",
    "        if line.startswith('>'):\n",
    "            fasta_name = line.rstrip()\n",
    "        else:\n",
    "            fasta_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0180ed2c",
   "metadata": {},
   "source": [
    "### Add data about clusters of gene expression for homeobox genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b21ffd46",
   "metadata": {},
   "source": [
    "Load annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "637b2118",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "      <th>hmmsearch_pfam_domains</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Pdum_DN1243_c0_g1_i44</td>\n",
       "      <td>6412.HelroP155469</td>\n",
       "      <td>5.1e-161</td>\n",
       "      <td>474.0</td>\n",
       "      <td>KOG0773@1|root,KOG0773@2759|Eukaryota,38E6H@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>regulation of cardiac muscle myoblast prolifer...</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>GO:0000122,GO:0000976,GO:0000977,GO:0000978,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox_KN,Meis_PKNOX_N</td>\n",
       "      <td>Homeobox_KN, Homeodomain, Meis_PKNOX_N</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pdum_DN13243_c0_g1_i14</td>\n",
       "      <td>126957.SMAR003960-PA</td>\n",
       "      <td>6.48e-137</td>\n",
       "      <td>410.0</td>\n",
       "      <td>KOG0773@1|root,KOG0773@2759|Eukaryota,38GSB@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>N-terminal of Homeobox Meis and PKNOX1</td>\n",
       "      <td>PKNOX2</td>\n",
       "      <td>GO:0000976,GO:0000977,GO:0000981,GO:0001012,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox_KN,Meis_PKNOX_N</td>\n",
       "      <td>Homeobox_KN, Homeodomain, Meis_PKNOX_N</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pdum_DN13531_c0_g1_i16</td>\n",
       "      <td>7739.XP_002609066.1</td>\n",
       "      <td>1.73e-77</td>\n",
       "      <td>244.0</td>\n",
       "      <td>KOG0488@1|root,KOG0488@2759|Eukaryota,38GM0@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>negative regulation of dendrite morphogenesis</td>\n",
       "      <td>TLX3</td>\n",
       "      <td>GO:0000981,GO:0001228,GO:0001704,GO:0001707,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox</td>\n",
       "      <td>Homeodomain</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pdum_DN13559_c0_g1_i20</td>\n",
       "      <td>126957.SMAR006755-PA</td>\n",
       "      <td>2.93e-39</td>\n",
       "      <td>147.0</td>\n",
       "      <td>KOG0490@1|root,KOG0490@2759|Eukaryota,3A22W@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>sequence-specific DNA binding. It is involved ...</td>\n",
       "      <td>repo</td>\n",
       "      <td>GO:0000976,GO:0000977,GO:0000981,GO:0000982,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox,OAR</td>\n",
       "      <td>Homeobox_KN, Homeodomain, OAR</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Pdum_DN13559_c0_g1_i8</td>\n",
       "      <td>126957.SMAR006755-PA</td>\n",
       "      <td>2.93e-39</td>\n",
       "      <td>147.0</td>\n",
       "      <td>KOG0490@1|root,KOG0490@2759|Eukaryota,3A22W@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>sequence-specific DNA binding. It is involved ...</td>\n",
       "      <td>repo</td>\n",
       "      <td>GO:0000976,GO:0000977,GO:0000981,GO:0000982,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox,OAR</td>\n",
       "      <td>Homeobox_KN, Homeodomain, OAR</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    query         seed_ortholog     evalue  score  \\\n",
       "0   Pdum_DN1243_c0_g1_i44     6412.HelroP155469   5.1e-161  474.0   \n",
       "1  Pdum_DN13243_c0_g1_i14  126957.SMAR003960-PA  6.48e-137  410.0   \n",
       "2  Pdum_DN13531_c0_g1_i16   7739.XP_002609066.1   1.73e-77  244.0   \n",
       "3  Pdum_DN13559_c0_g1_i20  126957.SMAR006755-PA   2.93e-39  147.0   \n",
       "4   Pdum_DN13559_c0_g1_i8  126957.SMAR006755-PA   2.93e-39  147.0   \n",
       "\n",
       "                                          eggNOG_OGs  max_annot_lvl  \\\n",
       "0  KOG0773@1|root,KOG0773@2759|Eukaryota,38E6H@33...  33208|Metazoa   \n",
       "1  KOG0773@1|root,KOG0773@2759|Eukaryota,38GSB@33...  33208|Metazoa   \n",
       "2  KOG0488@1|root,KOG0488@2759|Eukaryota,38GM0@33...  33208|Metazoa   \n",
       "3  KOG0490@1|root,KOG0490@2759|Eukaryota,3A22W@33...  33208|Metazoa   \n",
       "4  KOG0490@1|root,KOG0490@2759|Eukaryota,3A22W@33...  33208|Metazoa   \n",
       "\n",
       "  COG_category                                        Description  \\\n",
       "0            K  regulation of cardiac muscle myoblast prolifer...   \n",
       "1            K             N-terminal of Homeobox Meis and PKNOX1   \n",
       "2            K      negative regulation of dendrite morphogenesis   \n",
       "3            K  sequence-specific DNA binding. It is involved ...   \n",
       "4            K  sequence-specific DNA binding. It is involved ...   \n",
       "\n",
       "  Preferred_name                                                GOs  ...  \\\n",
       "0          MEIS2  GO:0000122,GO:0000976,GO:0000977,GO:0000978,GO...  ...   \n",
       "1         PKNOX2  GO:0000976,GO:0000977,GO:0000981,GO:0001012,GO...  ...   \n",
       "2           TLX3  GO:0000981,GO:0001228,GO:0001704,GO:0001707,GO...  ...   \n",
       "3           repo  GO:0000976,GO:0000977,GO:0000981,GO:0000982,GO...  ...   \n",
       "4           repo  GO:0000976,GO:0000977,GO:0000981,GO:0000982,GO...  ...   \n",
       "\n",
       "  KEGG_Module KEGG_Reaction KEGG_rclass                    BRITE KEGG_TC CAZy  \\\n",
       "0           -             -           -  ko00000,ko00001,ko03000       -    -   \n",
       "1           -             -           -                        -       -    -   \n",
       "2           -             -           -  ko00000,ko00001,ko03000       -    -   \n",
       "3           -             -           -  ko00000,ko00001,ko03000       -    -   \n",
       "4           -             -           -  ko00000,ko00001,ko03000       -    -   \n",
       "\n",
       "  BiGG_Reaction                     PFAMs  \\\n",
       "0             -  Homeobox_KN,Meis_PKNOX_N   \n",
       "1             -  Homeobox_KN,Meis_PKNOX_N   \n",
       "2             -                  Homeobox   \n",
       "3             -              Homeobox,OAR   \n",
       "4             -              Homeobox,OAR   \n",
       "\n",
       "                   hmmsearch_pfam_domains clusters  \n",
       "0  Homeobox_KN, Homeodomain, Meis_PKNOX_N           \n",
       "1  Homeobox_KN, Homeodomain, Meis_PKNOX_N           \n",
       "2                             Homeodomain           \n",
       "3           Homeobox_KN, Homeodomain, OAR           \n",
       "4           Homeobox_KN, Homeodomain, OAR           \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations = pd.read_csv\\\n",
    "    (pathlib.Path(r'C:\\worms\\new\\combined_annotations\\Pdum_homeobox_annotations.csv'), sep='\\t')\n",
    "Pdum_homeobox_annotations['clusters'] = ''\n",
    "Pdum_homeobox_annotations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "45155bb9",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>seed_ortholog</th>\n",
       "      <th>evalue</th>\n",
       "      <th>score</th>\n",
       "      <th>eggNOG_OGs</th>\n",
       "      <th>max_annot_lvl</th>\n",
       "      <th>COG_category</th>\n",
       "      <th>Description</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>GOs</th>\n",
       "      <th>...</th>\n",
       "      <th>KEGG_Module</th>\n",
       "      <th>KEGG_Reaction</th>\n",
       "      <th>KEGG_rclass</th>\n",
       "      <th>BRITE</th>\n",
       "      <th>KEGG_TC</th>\n",
       "      <th>CAZy</th>\n",
       "      <th>BiGG_Reaction</th>\n",
       "      <th>PFAMs</th>\n",
       "      <th>hmmsearch_pfam_domains</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Pele_DN137_c0_g2_i1</td>\n",
       "      <td>7955.ENSDARP00000117693</td>\n",
       "      <td>1.11e-38</td>\n",
       "      <td>149.0</td>\n",
       "      <td>KOG0489@1|root,KOG0842@2759|Eukaryota,39P9G@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>NK2 transcription factor related 2a</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>GO:0000122,GO:0000902,GO:0000904,GO:0000976,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox</td>\n",
       "      <td>Homeodomain</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pele_DN137_c0_g2_i3</td>\n",
       "      <td>6500.XP_005089677.1</td>\n",
       "      <td>7.66e-130</td>\n",
       "      <td>417.0</td>\n",
       "      <td>KOG0800@1|root,KOG0800@2759|Eukaryota,38DHK@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>O</td>\n",
       "      <td>Zinc finger, SWIM-type containing 2</td>\n",
       "      <td>ZSWIM2</td>\n",
       "      <td>GO:0000209,GO:0003674,GO:0003824,GO:0004842,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko01000,ko04030,ko04121</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox,SWIM,ZZ,zf-RING_2</td>\n",
       "      <td>Homeodomain, SWIM, zf-C3HC4, zf-C3HC4_2, zf-rb...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pele_DN137_c0_g2_i7</td>\n",
       "      <td>126957.SMAR012980-PA</td>\n",
       "      <td>2.64e-38</td>\n",
       "      <td>147.0</td>\n",
       "      <td>KOG0489@1|root,KOG0842@2759|Eukaryota,39P9G@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>sequence-specific DNA binding. It is involved ...</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>GO:0000122,GO:0000902,GO:0000904,GO:0000976,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko00001,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox</td>\n",
       "      <td>Homeodomain</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pele_DN13913_c0_g1_i11</td>\n",
       "      <td>7070.TC001291-PA</td>\n",
       "      <td>7.659999999999999e-81</td>\n",
       "      <td>261.0</td>\n",
       "      <td>KOG0490@1|root,KOG0490@2759|Eukaryota,396MF@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>binding. It is involved in the biological proc...</td>\n",
       "      <td>-</td>\n",
       "      <td>GO:0000003,GO:0000902,GO:0000904,GO:0001505,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox,LIM</td>\n",
       "      <td>Homeobox_KN, Homeodomain, LIM</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Pele_DN1472_c0_g1_i1</td>\n",
       "      <td>6412.HelroP76830</td>\n",
       "      <td>3.2800000000000003e-43</td>\n",
       "      <td>152.0</td>\n",
       "      <td>KOG0489@1|root,KOG0489@2759|Eukaryota,38B8X@33...</td>\n",
       "      <td>33208|Metazoa</td>\n",
       "      <td>K</td>\n",
       "      <td>DNA binding</td>\n",
       "      <td>HOXB7</td>\n",
       "      <td>GO:0000003,GO:0000976,GO:0000977,GO:0000978,GO...</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ko00000,ko03000</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Homeobox</td>\n",
       "      <td>Homeodomain</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    query            seed_ortholog                  evalue  \\\n",
       "0     Pele_DN137_c0_g2_i1  7955.ENSDARP00000117693                1.11e-38   \n",
       "1     Pele_DN137_c0_g2_i3      6500.XP_005089677.1               7.66e-130   \n",
       "2     Pele_DN137_c0_g2_i7     126957.SMAR012980-PA                2.64e-38   \n",
       "3  Pele_DN13913_c0_g1_i11         7070.TC001291-PA   7.659999999999999e-81   \n",
       "4    Pele_DN1472_c0_g1_i1         6412.HelroP76830  3.2800000000000003e-43   \n",
       "\n",
       "   score                                         eggNOG_OGs  max_annot_lvl  \\\n",
       "0  149.0  KOG0489@1|root,KOG0842@2759|Eukaryota,39P9G@33...  33208|Metazoa   \n",
       "1  417.0  KOG0800@1|root,KOG0800@2759|Eukaryota,38DHK@33...  33208|Metazoa   \n",
       "2  147.0  KOG0489@1|root,KOG0842@2759|Eukaryota,39P9G@33...  33208|Metazoa   \n",
       "3  261.0  KOG0490@1|root,KOG0490@2759|Eukaryota,396MF@33...  33208|Metazoa   \n",
       "4  152.0  KOG0489@1|root,KOG0489@2759|Eukaryota,38B8X@33...  33208|Metazoa   \n",
       "\n",
       "  COG_category                                        Description  \\\n",
       "0            K                NK2 transcription factor related 2a   \n",
       "1            O                Zinc finger, SWIM-type containing 2   \n",
       "2            K  sequence-specific DNA binding. It is involved ...   \n",
       "3            K  binding. It is involved in the biological proc...   \n",
       "4            K                                        DNA binding   \n",
       "\n",
       "  Preferred_name                                                GOs  ...  \\\n",
       "0         NKX2-2  GO:0000122,GO:0000902,GO:0000904,GO:0000976,GO...  ...   \n",
       "1         ZSWIM2  GO:0000209,GO:0003674,GO:0003824,GO:0004842,GO...  ...   \n",
       "2         NKX2-2  GO:0000122,GO:0000902,GO:0000904,GO:0000976,GO...  ...   \n",
       "3              -  GO:0000003,GO:0000902,GO:0000904,GO:0001505,GO...  ...   \n",
       "4          HOXB7  GO:0000003,GO:0000976,GO:0000977,GO:0000978,GO...  ...   \n",
       "\n",
       "  KEGG_Module KEGG_Reaction KEGG_rclass  \\\n",
       "0           -             -           -   \n",
       "1           -             -           -   \n",
       "2           -             -           -   \n",
       "3           -             -           -   \n",
       "4           -             -           -   \n",
       "\n",
       "                                     BRITE KEGG_TC CAZy BiGG_Reaction  \\\n",
       "0                  ko00000,ko00001,ko03000       -    -             -   \n",
       "1  ko00000,ko00001,ko01000,ko04030,ko04121       -    -             -   \n",
       "2                  ko00000,ko00001,ko03000       -    -             -   \n",
       "3                          ko00000,ko03000       -    -             -   \n",
       "4                          ko00000,ko03000       -    -             -   \n",
       "\n",
       "                        PFAMs  \\\n",
       "0                    Homeobox   \n",
       "1  Homeobox,SWIM,ZZ,zf-RING_2   \n",
       "2                    Homeobox   \n",
       "3                Homeobox,LIM   \n",
       "4                    Homeobox   \n",
       "\n",
       "                              hmmsearch_pfam_domains clusters  \n",
       "0                                        Homeodomain           \n",
       "1  Homeodomain, SWIM, zf-C3HC4, zf-C3HC4_2, zf-rb...           \n",
       "2                                        Homeodomain           \n",
       "3                      Homeobox_KN, Homeodomain, LIM           \n",
       "4                                        Homeodomain           \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations = pd.read_csv\\\n",
    "    (pathlib.Path(r'C:\\worms\\new\\combined_annotations\\Pele_homeobox_annotations.csv'), sep='\\t')\n",
    "Pele_homeobox_annotations['clusters'] = ''\n",
    "Pele_homeobox_annotations.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccd20b27",
   "metadata": {},
   "source": [
    "Load data about gene expression clusters from Clust"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "351ac82e",
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters_homeo_df = {}\n",
    "p = pathlib.Path(r'C:\\worms\\new\\combined_annotations\\cluster_info').glob('**/*')\n",
    "files = [x for x in p if x.is_file()]\n",
    "for cluter_file in files:\n",
    "    clusters_homeo_df[cluter_file.stem.rstrip('_hom')] = pd.read_csv(cluter_file, sep=',')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "152f3d2c",
   "metadata": {},
   "source": [
    "Combine expression data and annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d6b7b5d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "for species_site in clusters_homeo_df.keys():\n",
    "    species, site = species_site.split('_')\n",
    "    for column in clusters_homeo_df[species_site].columns:\n",
    "        site_cluster = site+'_'+column\n",
    "        if species == 'pdum':\n",
    "            Pdum_homeobox_annotations.loc\\\n",
    "                [Pdum_homeobox_annotations['query'].isin(list(clusters_homeo_df[species_site][column].dropna())),'clusters'] += site_cluster\n",
    "            Pdum_homeobox_annotations[site_cluster]\\\n",
    "                = Pdum_homeobox_annotations['query'].isin(list(clusters_homeo_df[species_site][column].dropna()))\n",
    "                \n",
    "        elif species == 'pel':\n",
    "            Pele_homeobox_annotations.loc\\\n",
    "                [Pele_homeobox_annotations['query'].isin(list(clusters_homeo_df[species_site][column].dropna())),'clusters'] += site_cluster\n",
    "            Pele_homeobox_annotations[site_cluster]\\\n",
    "                = Pele_homeobox_annotations['query'].isin(list(clusters_homeo_df[species_site][column].dropna()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60ff100d",
   "metadata": {},
   "source": [
    "### Let's explore clusters with homeobox genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c18b910b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NKX2-2', 'ZSWIM2', '-', 'HOXB7', 'VAX1', 'PBX1', 'pbx4', 'MSX2',\n",
       "       'IRX3', 'MKX', 'ISL1', 'MEIS2', 'NKX2-3', 'HOXD12', 'PHTF1',\n",
       "       'DLX1', 'ONECUT2', 'HOXB4', 'NKX2-1', 'PKNOX2', 'PDX1', 'POU6F2',\n",
       "       'PRRX2', 'MNX1', 'PROP1', 'EVX2', 'MEOX2', 'ISX', 'NKX3-2',\n",
       "       'POU4F2'], dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations['Preferred_name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "86fd59bf",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['MEIS2', 'PKNOX2', 'TLX3', 'repo', 'EVX1', 'LHX1', 'CERS6',\n",
       "       'NKX6-1', 'HHEX', 'HOXB7', 'NKX2-1', 'POU6F2', 'EN2', 'LBX1', '-',\n",
       "       'UNCX', 'HLX', 'HOXA5', 'HOXB2', 'NKX2-3', 'IRX4', 'NKX1-1',\n",
       "       'ISL1', 'ZEB2', 'NKX2-2', 'PHTF1', 'cad', 'IPO13', 'MKX', 'dve-1',\n",
       "       'SIX2', 'CUX2', 'DLX1', 'PRRX1', 'EMX2', 'POU4F2', 'TGIF1',\n",
       "       'PITX1', 'PBX1', 'MSX2', 'HOXA7', 'HOXB4', 'MNX1', 'ONECUT1',\n",
       "       'CERS5'], dtype=object)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations['Preferred_name'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac9d612f",
   "metadata": {},
   "source": [
    "Clusters with homeobox genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9c5f3e12",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pdum_head\n",
      "C0     True\n",
      "C1     True\n",
      "C2     True\n",
      "C3    False\n",
      "C4    False\n",
      "C5    False\n",
      "C6    False\n",
      "C7     True\n",
      "C8    False\n",
      "dtype: bool\n",
      "pdum_tail\n",
      "C0     True\n",
      "C1     True\n",
      "C2     True\n",
      "C3    False\n",
      "dtype: bool\n",
      "pel_head\n",
      "C0     True\n",
      "C1    False\n",
      "C2    False\n",
      "C3     True\n",
      "C4     True\n",
      "C5     True\n",
      "C6     True\n",
      "C7    False\n",
      "dtype: bool\n",
      "pel_tail\n",
      "C0     True\n",
      "C1     True\n",
      "C2    False\n",
      "C3     True\n",
      "C4     True\n",
      "C5    False\n",
      "C6     True\n",
      "C7     True\n",
      "dtype: bool\n"
     ]
    }
   ],
   "source": [
    "for species_site, df in clusters_homeo_df.items():\n",
    "    print(species_site)\n",
    "    print(df.isna().sum() != df.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5145e72d",
   "metadata": {},
   "source": [
    "HOX7-like, NKX-like and PBX-like genes show interesting patterns of expression after clustering, some of them expressed only during anterior regeneration (tail) or posterior regeneration (head)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6b5856d4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>query</th>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Pdum_DN18748_c0_g1_i2</th>\n",
       "      <th>NKX6-1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_DN2241_c0_g1_i3</th>\n",
       "      <th>NKX2-1</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_DN31695_c0_g1_i14</th>\n",
       "      <th>NKX2-3</th>\n",
       "      <th>head_C0tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_DN51214_c0_g1_i6</th>\n",
       "      <th>NKX1-1</th>\n",
       "      <th>head_C2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_DN5666_c1_g1_i1</th>\n",
       "      <th>NKX2-2</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_NODE_60345_length_2007_cov_132_810133_g16879_i0</th>\n",
       "      <th>NKX2-2</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdum_NODE_63868_length_1925_cov_20_905449_g17901_i0</th>\n",
       "      <th>NKX1-1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                  query\n",
       "query                                              Preferred_name clusters             \n",
       "Pdum_DN18748_c0_g1_i2                              NKX6-1                             1\n",
       "Pdum_DN2241_c0_g1_i3                               NKX2-1         head_C0             1\n",
       "Pdum_DN31695_c0_g1_i14                             NKX2-3         head_C0tail_C0      1\n",
       "Pdum_DN51214_c0_g1_i6                              NKX1-1         head_C2             1\n",
       "Pdum_DN5666_c1_g1_i1                               NKX2-2         head_C0             1\n",
       "Pdum_NODE_60345_length_2007_cov_132_810133_g168... NKX2-2         head_C0             1\n",
       "Pdum_NODE_63868_length_1925_cov_20_905449_g1790... NKX1-1                             1"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations[Pdum_homeobox_annotations['Preferred_name'].str.contains('NKX')]\\\n",
    "    .groupby(['query','Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "c6a4d6cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">PBX1</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         query\n",
       "Preferred_name clusters       \n",
       "PBX1                         2\n",
       "               head_C0       1"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations[Pdum_homeobox_annotations['Preferred_name'].str.contains('pbx', case=False)]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "86969490",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">PBX1</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C5tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C1</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pbx4</th>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "PBX1                               2\n",
       "               head_C5tail_C1      1\n",
       "               tail_C1             2\n",
       "pbx4           tail_C0             1"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations[Pele_homeobox_annotations['Preferred_name'].str.contains('pbx', case=False)]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "a947bd56",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HOXA7</th>\n",
       "      <th>head_C1tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">HOXB7</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C0</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "HOXA7          head_C1tail_C1      1\n",
       "HOXB7                              2\n",
       "               head_C0             4"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations[Pdum_homeobox_annotations['Preferred_name'].str.contains('hox.*7', case=False)]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "1a00458b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">HOXB7</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C3tail_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "HOXB7                              1\n",
       "               head_C3tail_C4      1\n",
       "               tail_C1             1\n",
       "               tail_C4             1"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations[Pele_homeobox_annotations['Preferred_name'].str.contains('hox.*7', case=False)]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "c9107230",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NKX2-1</th>\n",
       "      <th></th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NKX2-2</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-3</th>\n",
       "      <th>head_C3tail_C7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX3-2</th>\n",
       "      <th>head_C5tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "NKX2-1                             3\n",
       "NKX2-2                             1\n",
       "               tail_C7             1\n",
       "NKX2-3         head_C3tail_C7      1\n",
       "NKX3-2         head_C5tail_C0      1"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations[Pele_homeobox_annotations['Preferred_name'].str.contains('NKX')]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "df1e8963",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NKX1-1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C2</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-1</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-2</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-3</th>\n",
       "      <th>head_C0tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX6-1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "NKX1-1                             1\n",
       "               head_C2             1\n",
       "NKX2-1         head_C0             1\n",
       "NKX2-2         head_C0             2\n",
       "NKX2-3         head_C0tail_C0      1\n",
       "NKX6-1                             1"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pdum_homeobox_annotations[Pdum_homeobox_annotations['Preferred_name'].str.contains('NKX')]\\\n",
    "    .groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "14219a47",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Preferred_name</th>\n",
       "      <th>clusters</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">-</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DLX1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EVX2</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXB4</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">HOXB7</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C3tail_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOXD12</th>\n",
       "      <th>head_C3tail_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IRX3</th>\n",
       "      <th>head_C6tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ISL1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ISX</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEIS2</th>\n",
       "      <th></th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MEOX2</th>\n",
       "      <th>head_C5tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">MKX</th>\n",
       "      <th>tail_C3</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C6</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MNX1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">MSX2</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-1</th>\n",
       "      <th></th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">NKX2-2</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-3</th>\n",
       "      <th>head_C3tail_C7</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX3-2</th>\n",
       "      <th>head_C5tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ONECUT2</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">PBX1</th>\n",
       "      <th></th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C5tail_C1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tail_C1</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDX1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHTF1</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PKNOX2</th>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">POU4F2</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>head_C5</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POU6F2</th>\n",
       "      <th>head_C4</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PROP1</th>\n",
       "      <th>head_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRRX2</th>\n",
       "      <th>head_C4tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VAX1</th>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZSWIM2</th>\n",
       "      <th></th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pbx4</th>\n",
       "      <th>tail_C0</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               query\n",
       "Preferred_name clusters             \n",
       "-                                  2\n",
       "               head_C4             1\n",
       "               tail_C0             1\n",
       "DLX1                               1\n",
       "EVX2                               2\n",
       "HOXB4                              1\n",
       "HOXB7                              1\n",
       "               head_C3tail_C4      1\n",
       "               tail_C1             1\n",
       "               tail_C4             1\n",
       "HOXD12         head_C3tail_C4      1\n",
       "IRX3           head_C6tail_C1      1\n",
       "ISL1                               1\n",
       "ISX                                2\n",
       "MEIS2                              5\n",
       "MEOX2          head_C5tail_C0      1\n",
       "MKX            tail_C3             2\n",
       "               tail_C6             1\n",
       "MNX1                               1\n",
       "MSX2                               1\n",
       "               tail_C0             1\n",
       "NKX2-1                             3\n",
       "NKX2-2                             1\n",
       "               tail_C7             1\n",
       "NKX2-3         head_C3tail_C7      1\n",
       "NKX3-2         head_C5tail_C0      1\n",
       "ONECUT2                            2\n",
       "PBX1                               2\n",
       "               head_C5tail_C1      1\n",
       "               tail_C1             2\n",
       "PDX1                               1\n",
       "PHTF1                              1\n",
       "PKNOX2         tail_C0             1\n",
       "POU4F2                             1\n",
       "               head_C5             1\n",
       "POU6F2         head_C4             1\n",
       "PROP1          head_C0             1\n",
       "PRRX2          head_C4tail_C0      1\n",
       "VAX1           tail_C0             1\n",
       "ZSWIM2                             1\n",
       "pbx4           tail_C0             1"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pele_homeobox_annotations.groupby(['Preferred_name','clusters']).agg({'query':'count'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a250e58",
   "metadata": {},
   "source": [
    "saving combined annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e60b0c2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Pdum_homeobox_annotations.to_csv('C:/worms/new/combined_annotations/pdum_homeo_clusters_added.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "05fd5d4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "Pele_homeobox_annotations.to_csv('C:/worms/new/combined_annotations/pele_homeo_clusters_added.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4919e46f",
   "metadata": {},
   "source": [
    "### Extracting orthogroups with homeobox genes of interesting from proteinortho results for further phylogenetic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0ad79b4c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Species</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Alg.-Conn.</th>\n",
       "      <th>Apime_pep.fasta</th>\n",
       "      <th>Biogl_pep.fasta</th>\n",
       "      <th>Capte_pep.fasta</th>\n",
       "      <th>Cragi_pep.fasta</th>\n",
       "      <th>Dimgy_pep.fasta</th>\n",
       "      <th>Helro_pep.fasta</th>\n",
       "      <th>Lotgi_pep.fasta</th>\n",
       "      <th>Myted_pep.fasta</th>\n",
       "      <th>Owefu_pep.fasta</th>\n",
       "      <th>Pdum_amin_av.fasta</th>\n",
       "      <th>Pele_amin_av.fasta</th>\n",
       "      <th>Pomca_pep.fasta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>79</td>\n",
       "      <td>0.130</td>\n",
       "      <td>tr|A0A7M7TEW6|A0A7M7TEW6_APIME,tr|A0A7M7TF59|A...</td>\n",
       "      <td>tr|A0A2C9JVG7|A0A2C9JVG7_BIOGL,tr|A0A2C9K3W4|A...</td>\n",
       "      <td>tr|R7VBM0|R7VBM0_CAPTE,tr|R7TKU8|R7TKU8_CAPTE,...</td>\n",
       "      <td>tr|K1PN21|K1PN21_CRAGI,tr|K1R7V7|K1R7V7_CRAGI,...</td>\n",
       "      <td>tr|A0A7I8V5G4|A0A7I8V5G4_9ANNE,tr|A0A7I8VRA9|A...</td>\n",
       "      <td>tr|T1FP49|T1FP49_HELRO,tr|T1G8V0|T1G8V0_HELRO,...</td>\n",
       "      <td>tr|V4AK06|V4AK06_LOTGI,tr|V4AP80|V4AP80_LOTGI,...</td>\n",
       "      <td>tr|A0A8S3RWK7|A0A8S3RWK7_MYTED,tr|A0A8S3QXJ8|A...</td>\n",
       "      <td>tr|A0A8J1XSR5|A0A8J1XSR5_OWEFU,tr|A0A8J1TCI4|A...</td>\n",
       "      <td>Pdum_NODE_70038_length_1792_cov_368_286947_g19...</td>\n",
       "      <td>Pele_NODE_65009_length_1834_cov_568_581488_g28...</td>\n",
       "      <td>tr|A0A2T7NK64|A0A2T7NK64_POMCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12</td>\n",
       "      <td>66</td>\n",
       "      <td>0.258</td>\n",
       "      <td>tr|A0A7M6UW32|A0A7M6UW32_APIME,tr|A0A7M7MKH4|A...</td>\n",
       "      <td>sp|P92179|ACTC_BIOGL,tr|A0A2C9JHE8|A0A2C9JHE8_...</td>\n",
       "      <td>tr|R7TPD2|R7TPD2_CAPTE,tr|R7TMG3|R7TMG3_CAPTE</td>\n",
       "      <td>tr|A0A8W8K574|A0A8W8K574_CRAGI,tr|A0A8W8M4F9|A...</td>\n",
       "      <td>tr|A0A7I8W774|A0A7I8W774_9ANNE,tr|A0A7I8VAG7|A...</td>\n",
       "      <td>tr|T1FQQ4|T1FQQ4_HELRO,tr|T1FMD2|T1FMD2_HELRO,...</td>\n",
       "      <td>tr|V4BC69|V4BC69_LOTGI,tr|V4B4D3|V4B4D3_LOTGI,...</td>\n",
       "      <td>tr|A0A8S3QH94|A0A8S3QH94_MYTED,tr|A0A8S3TVQ9|A...</td>\n",
       "      <td>tr|A0A8J1XVJ4|A0A8J1XVJ4_OWEFU,tr|A0A8J1UKN2|A...</td>\n",
       "      <td>Pdum_NODE_64838_length_1903_cov_75_402703_g146...</td>\n",
       "      <td>Pele_DN43937_c0_g1_i16,Pele_NODE_104175_length...</td>\n",
       "      <td>tr|A0A2T7PW49|A0A2T7PW49_POMCA,tr|A0A2T7NL57|A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12</td>\n",
       "      <td>63</td>\n",
       "      <td>0.116</td>\n",
       "      <td>tr|A0A7M7IKC6|A0A7M7IKC6_APIME,tr|A0A7M7R7R1|A...</td>\n",
       "      <td>tr|A0A2C9JGA9|A0A2C9JGA9_BIOGL,tr|A0A2C9JQK4|A...</td>\n",
       "      <td>tr|R7TF87|R7TF87_CAPTE,tr|R7TZN1|R7TZN1_CAPTE</td>\n",
       "      <td>tr|K1QC26|K1QC26_CRAGI,tr|A0A8W8NNG6|A0A8W8NNG...</td>\n",
       "      <td>tr|A0A7I8W7C6|A0A7I8W7C6_9ANNE,tr|A0A7I8W471|A...</td>\n",
       "      <td>tr|T1G3P4|T1G3P4_HELRO,tr|T1G3S4|T1G3S4_HELRO,...</td>\n",
       "      <td>tr|V4CC90|V4CC90_LOTGI,tr|V4A629|V4A629_LOTGI,...</td>\n",
       "      <td>tr|A0A8S3QPR1|A0A8S3QPR1_MYTED,tr|A0A8S3Q770|A...</td>\n",
       "      <td>tr|A0A8J1U5H0|A0A8J1U5H0_OWEFU,tr|A0A8S4QDI4|A...</td>\n",
       "      <td>Pdum_DN212_c0_g1_i11,Pdum_DN651_c0_g1_i12,Pdum...</td>\n",
       "      <td>Pele_DN11477_c0_g1_i7,Pele_DN11477_c0_g1_i13,P...</td>\n",
       "      <td>tr|A0A2T7NST7|A0A2T7NST7_POMCA,tr|A0A2T7Q019|A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12</td>\n",
       "      <td>55</td>\n",
       "      <td>0.116</td>\n",
       "      <td>tr|A0A7M7H2E2|A0A7M7H2E2_APIME,tr|A0A7M7L4P4|A...</td>\n",
       "      <td>tr|A0A2C9JRZ8|A0A2C9JRZ8_BIOGL,tr|A0A2C9JS03|A...</td>\n",
       "      <td>tr|R7T3F9|R7T3F9_CAPTE</td>\n",
       "      <td>tr|A0A8W8HKV7|A0A8W8HKV7_CRAGI,tr|A0A8W8HLG3|A...</td>\n",
       "      <td>tr|A0A7I8W6A3|A0A7I8W6A3_9ANNE,tr|A0A7I8W6H0|A...</td>\n",
       "      <td>tr|T1EET5|T1EET5_HELRO</td>\n",
       "      <td>tr|V3ZMJ2|V3ZMJ2_LOTGI</td>\n",
       "      <td>tr|A0A8S3Q3J9|A0A8S3Q3J9_MYTED,tr|A0A8S3Q2D1|A...</td>\n",
       "      <td>tr|A0A8S4PW64|A0A8S4PW64_OWEFU,tr|A0A8J1Y8X9|A...</td>\n",
       "      <td>Pdum_DN1549_c0_g1_i19,Pdum_DN1549_c0_g1_i17,Pd...</td>\n",
       "      <td>Pele_DN42817_c0_g1_i17,Pele_NODE_9304_length_4...</td>\n",
       "      <td>tr|A0A2T7NEX5|A0A2T7NEX5_POMCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>50</td>\n",
       "      <td>0.141</td>\n",
       "      <td>tr|A0A7M7SQ67|A0A7M7SQ67_APIME,tr|A0A7M7SQF4|A...</td>\n",
       "      <td>tr|A0A2C9JFZ1|A0A2C9JFZ1_BIOGL</td>\n",
       "      <td>tr|R7UY67|R7UY67_CAPTE</td>\n",
       "      <td>tr|A0A8W8MVX4|A0A8W8MVX4_CRAGI,tr|K1R5S7|K1R5S...</td>\n",
       "      <td>tr|A0A7I8VF00|A0A7I8VF00_9ANNE</td>\n",
       "      <td>tr|T1FXX2|T1FXX2_HELRO</td>\n",
       "      <td>tr|V4B1X9|V4B1X9_LOTGI</td>\n",
       "      <td>tr|A0A8S3SHN4|A0A8S3SHN4_MYTED,tr|A0A8S3SK91|A...</td>\n",
       "      <td>tr|A0A8J1U4R7|A0A8J1U4R7_OWEFU,tr|A0A8J1U698|A...</td>\n",
       "      <td>Pdum_DN1818_c0_g1_i11,Pdum_NODE_4564_length_61...</td>\n",
       "      <td>Pele_DN16530_c1_g1_i13,Pele_NODE_9278_length_4...</td>\n",
       "      <td>tr|A0A2T7PWY1|A0A2T7PWY1_POMCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   # Species  Genes  Alg.-Conn.  \\\n",
       "0         12     79       0.130   \n",
       "1         12     66       0.258   \n",
       "2         12     63       0.116   \n",
       "3         12     55       0.116   \n",
       "4         12     50       0.141   \n",
       "\n",
       "                                     Apime_pep.fasta  \\\n",
       "0  tr|A0A7M7TEW6|A0A7M7TEW6_APIME,tr|A0A7M7TF59|A...   \n",
       "1  tr|A0A7M6UW32|A0A7M6UW32_APIME,tr|A0A7M7MKH4|A...   \n",
       "2  tr|A0A7M7IKC6|A0A7M7IKC6_APIME,tr|A0A7M7R7R1|A...   \n",
       "3  tr|A0A7M7H2E2|A0A7M7H2E2_APIME,tr|A0A7M7L4P4|A...   \n",
       "4  tr|A0A7M7SQ67|A0A7M7SQ67_APIME,tr|A0A7M7SQF4|A...   \n",
       "\n",
       "                                     Biogl_pep.fasta  \\\n",
       "0  tr|A0A2C9JVG7|A0A2C9JVG7_BIOGL,tr|A0A2C9K3W4|A...   \n",
       "1  sp|P92179|ACTC_BIOGL,tr|A0A2C9JHE8|A0A2C9JHE8_...   \n",
       "2  tr|A0A2C9JGA9|A0A2C9JGA9_BIOGL,tr|A0A2C9JQK4|A...   \n",
       "3  tr|A0A2C9JRZ8|A0A2C9JRZ8_BIOGL,tr|A0A2C9JS03|A...   \n",
       "4                     tr|A0A2C9JFZ1|A0A2C9JFZ1_BIOGL   \n",
       "\n",
       "                                     Capte_pep.fasta  \\\n",
       "0  tr|R7VBM0|R7VBM0_CAPTE,tr|R7TKU8|R7TKU8_CAPTE,...   \n",
       "1      tr|R7TPD2|R7TPD2_CAPTE,tr|R7TMG3|R7TMG3_CAPTE   \n",
       "2      tr|R7TF87|R7TF87_CAPTE,tr|R7TZN1|R7TZN1_CAPTE   \n",
       "3                             tr|R7T3F9|R7T3F9_CAPTE   \n",
       "4                             tr|R7UY67|R7UY67_CAPTE   \n",
       "\n",
       "                                     Cragi_pep.fasta  \\\n",
       "0  tr|K1PN21|K1PN21_CRAGI,tr|K1R7V7|K1R7V7_CRAGI,...   \n",
       "1  tr|A0A8W8K574|A0A8W8K574_CRAGI,tr|A0A8W8M4F9|A...   \n",
       "2  tr|K1QC26|K1QC26_CRAGI,tr|A0A8W8NNG6|A0A8W8NNG...   \n",
       "3  tr|A0A8W8HKV7|A0A8W8HKV7_CRAGI,tr|A0A8W8HLG3|A...   \n",
       "4  tr|A0A8W8MVX4|A0A8W8MVX4_CRAGI,tr|K1R5S7|K1R5S...   \n",
       "\n",
       "                                     Dimgy_pep.fasta  \\\n",
       "0  tr|A0A7I8V5G4|A0A7I8V5G4_9ANNE,tr|A0A7I8VRA9|A...   \n",
       "1  tr|A0A7I8W774|A0A7I8W774_9ANNE,tr|A0A7I8VAG7|A...   \n",
       "2  tr|A0A7I8W7C6|A0A7I8W7C6_9ANNE,tr|A0A7I8W471|A...   \n",
       "3  tr|A0A7I8W6A3|A0A7I8W6A3_9ANNE,tr|A0A7I8W6H0|A...   \n",
       "4                     tr|A0A7I8VF00|A0A7I8VF00_9ANNE   \n",
       "\n",
       "                                     Helro_pep.fasta  \\\n",
       "0  tr|T1FP49|T1FP49_HELRO,tr|T1G8V0|T1G8V0_HELRO,...   \n",
       "1  tr|T1FQQ4|T1FQQ4_HELRO,tr|T1FMD2|T1FMD2_HELRO,...   \n",
       "2  tr|T1G3P4|T1G3P4_HELRO,tr|T1G3S4|T1G3S4_HELRO,...   \n",
       "3                             tr|T1EET5|T1EET5_HELRO   \n",
       "4                             tr|T1FXX2|T1FXX2_HELRO   \n",
       "\n",
       "                                     Lotgi_pep.fasta  \\\n",
       "0  tr|V4AK06|V4AK06_LOTGI,tr|V4AP80|V4AP80_LOTGI,...   \n",
       "1  tr|V4BC69|V4BC69_LOTGI,tr|V4B4D3|V4B4D3_LOTGI,...   \n",
       "2  tr|V4CC90|V4CC90_LOTGI,tr|V4A629|V4A629_LOTGI,...   \n",
       "3                             tr|V3ZMJ2|V3ZMJ2_LOTGI   \n",
       "4                             tr|V4B1X9|V4B1X9_LOTGI   \n",
       "\n",
       "                                     Myted_pep.fasta  \\\n",
       "0  tr|A0A8S3RWK7|A0A8S3RWK7_MYTED,tr|A0A8S3QXJ8|A...   \n",
       "1  tr|A0A8S3QH94|A0A8S3QH94_MYTED,tr|A0A8S3TVQ9|A...   \n",
       "2  tr|A0A8S3QPR1|A0A8S3QPR1_MYTED,tr|A0A8S3Q770|A...   \n",
       "3  tr|A0A8S3Q3J9|A0A8S3Q3J9_MYTED,tr|A0A8S3Q2D1|A...   \n",
       "4  tr|A0A8S3SHN4|A0A8S3SHN4_MYTED,tr|A0A8S3SK91|A...   \n",
       "\n",
       "                                     Owefu_pep.fasta  \\\n",
       "0  tr|A0A8J1XSR5|A0A8J1XSR5_OWEFU,tr|A0A8J1TCI4|A...   \n",
       "1  tr|A0A8J1XVJ4|A0A8J1XVJ4_OWEFU,tr|A0A8J1UKN2|A...   \n",
       "2  tr|A0A8J1U5H0|A0A8J1U5H0_OWEFU,tr|A0A8S4QDI4|A...   \n",
       "3  tr|A0A8S4PW64|A0A8S4PW64_OWEFU,tr|A0A8J1Y8X9|A...   \n",
       "4  tr|A0A8J1U4R7|A0A8J1U4R7_OWEFU,tr|A0A8J1U698|A...   \n",
       "\n",
       "                                  Pdum_amin_av.fasta  \\\n",
       "0  Pdum_NODE_70038_length_1792_cov_368_286947_g19...   \n",
       "1  Pdum_NODE_64838_length_1903_cov_75_402703_g146...   \n",
       "2  Pdum_DN212_c0_g1_i11,Pdum_DN651_c0_g1_i12,Pdum...   \n",
       "3  Pdum_DN1549_c0_g1_i19,Pdum_DN1549_c0_g1_i17,Pd...   \n",
       "4  Pdum_DN1818_c0_g1_i11,Pdum_NODE_4564_length_61...   \n",
       "\n",
       "                                  Pele_amin_av.fasta  \\\n",
       "0  Pele_NODE_65009_length_1834_cov_568_581488_g28...   \n",
       "1  Pele_DN43937_c0_g1_i16,Pele_NODE_104175_length...   \n",
       "2  Pele_DN11477_c0_g1_i7,Pele_DN11477_c0_g1_i13,P...   \n",
       "3  Pele_DN42817_c0_g1_i17,Pele_NODE_9304_length_4...   \n",
       "4  Pele_DN16530_c1_g1_i13,Pele_NODE_9278_length_4...   \n",
       "\n",
       "                                     Pomca_pep.fasta  \n",
       "0                     tr|A0A2T7NK64|A0A2T7NK64_POMCA  \n",
       "1  tr|A0A2T7PW49|A0A2T7PW49_POMCA,tr|A0A2T7NL57|A...  \n",
       "2  tr|A0A2T7NST7|A0A2T7NST7_POMCA,tr|A0A2T7Q019|A...  \n",
       "3                     tr|A0A2T7NEX5|A0A2T7NEX5_POMCA  \n",
       "4                     tr|A0A2T7PWY1|A0A2T7PWY1_POMCA  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results_tsv_file = r'C:\\worms\\new\\proteinortho_results\\summaries\\12_species.proteinortho.tsv'\n",
    "proteinortho_df = pd.read_csv(results_tsv_file, sep='\\t')\n",
    "proteinortho_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8e81e821",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_orthologs(gene_name_pattern):\n",
    "    indices = []\n",
    "    pdum_genes = Pdum_homeobox_annotations\\\n",
    "        [Pdum_homeobox_annotations['Preferred_name'].str.contains(gene_name_pattern,case=False)]['query'].to_list()\n",
    "    for gene in pdum_genes:\n",
    "        if proteinortho_df[proteinortho_df['Pdum_amin_av.fasta'].str.contains(gene)].shape[0] != 0:\n",
    "            indices.append(proteinortho_df[proteinortho_df['Pdum_amin_av.fasta'].str.contains(gene)].index[0])\n",
    "            print(proteinortho_df[proteinortho_df['Pdum_amin_av.fasta'].str.contains(gene)].index[0])\n",
    "        else:\n",
    "            print('singletone')\n",
    "\n",
    "    pele_genes = Pele_homeobox_annotations\\\n",
    "        [Pele_homeobox_annotations['Preferred_name'].str.contains(gene_name_pattern,case=False)]['query'].to_list()\n",
    "    for gene in pele_genes:\n",
    "        if proteinortho_df[proteinortho_df['Pele_amin_av.fasta'].str.contains(gene)].shape[0] != 0:\n",
    "            indices.append(proteinortho_df[proteinortho_df['Pele_amin_av.fasta'].str.contains(gene)].index[0])\n",
    "            print(proteinortho_df[proteinortho_df['Pele_amin_av.fasta'].str.contains(gene)].index[0])\n",
    "        else:\n",
    "            print('singletone')\n",
    "\n",
    "    return ','.join([gene for gene in np.ravel(proteinortho_df.iloc[list(set(indices)),[3,4,5,6,7,8,9,10,11,14]].values) if gene != '*']\\\n",
    "        + pdum_genes\\\n",
    "        + pele_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4037e96",
   "metadata": {},
   "source": [
    "Saving genes from orthogroups of interest to extract fasta sequence with proteinortho grab_proteins.pl utility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d4025c06",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21869\n",
      "singletone\n",
      "14079\n",
      "singletone\n",
      "singletone\n",
      "15151\n",
      "singletone\n",
      "14079\n",
      "14947\n",
      "14947\n",
      "14079\n",
      "21\n",
      "21\n",
      "9595\n",
      "21\n",
      "21\n",
      "21\n",
      "singletone\n",
      "singletone\n",
      "21\n",
      "5595\n",
      "singletone\n",
      "30569\n",
      "4844\n",
      "singletone\n",
      "4844\n",
      "singletone\n",
      "singletone\n",
      "singletone\n",
      "5595\n"
     ]
    }
   ],
   "source": [
    "genes_of_interest = ['hox.*7','pbx','nkx2']\n",
    "for gene in genes_of_interest:\n",
    "    with open(pathlib.Path(r'C:\\worms\\new\\proteinortho_results\\orthogroups_for_phylogeny.txt'), 'a') as orthogroups_file:\n",
    "        orthogroups_file.write(gene+' '+get_all_orthologs(gene)+'\\n')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
